source/common/redis/command_splitter_impl.cc

```diff
@@ -52,10 +53,17 @@ void SimpleRequest::cancel() {
   handle_ = nullptr;
 }
 
+MGETRequest::~MGETRequest() {
+#ifndef NDEBUG
+  for (const PendingRequest& request : pending_requests_) {
```

Comment:
Prepare to be impressed. I compiled this with Clang++ -O2 -S to see the assembly (using cassert, but it should be equivalent to our macro):

```
void do_nothing_to_vector(std::vector<std::string>& foo) {
  for (const std::string& s : foo) {
    assert(s.size() > 1);
  }
}
```
If you add -DNDEBUG to the Clang command line the assembly it generates is just "retq"